### PR TITLE
Ship unfork

### PR DESCRIPTION
**Pull request by marccampbell**
What I Did
------------
Add a new "unfork" command to Ship. This takes two parameters: a URL to a forked/copied Helm chart, and a URL to an upstream Helm chart. Ship will run (in headless mode) to create Kustomize patches from the fork, so that the upstream helm chart can be used directly again. The values.yaml from the fork will be used when running Helm template against both charts.

How I Did it
------------
Modified Ship to support a new, default Unfork lifecycle. The "ship unfork" command always invokes this. This only works with 2 a forked Helm chart today.  Used the existing CreateTwoWayStrategicMergePatch functionality.

How to verify it
------------
Run the integration tests. Or:

```
ship unfork --upstream https://github.com/helm/charts/tree/353ba5ef6467fd64035b7d5446df426f86d60153/stable/grafana https://github.com/replicatedhq/test-charts/tree/2687f40789200c03c80df4831a56e9af77b2e701/grafana-forked
```

Description for the Changelog
------------
Add `ship unfork` command to create patches from an existing fork.

Picture of a Boat (not required but encouraged)
------------

clipart20572

### CODE REVIEW

File: integration/lib.go [hunk 1]
```diff
@@ -121,3 +136,64 @@ func CompareDir(expected, actual string, replacements map[string]string) (bool,
 
 	return true, nil
 }
+
+func prettyAndCleanJSON(data []byte, keysToIgnore []string) []byte {
```
Comment (ship-version-checker) [inline integration/lib.go]: do we still want to add this along with `modifyField`?

Comment (Rob0h) [inline integration/lib.go]: do we still want this and `modifyField` ?

File: pkg/lifecycle/unfork/daemonless.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+	State   state.Manager
+	Patcher patch.ShipPatcher
+}
+
+func NewDaemonlessUnforker(logger log.Logger, fs afero.Afero, state state.Manager) lifecycle.Unforker {
+	return &Unforker{
+		Logger: logger,
+		FS:     fs,
+		State:  state,
+		Patcher: patch.ShipPatcher{
```
Comment (Rob0h) [inline pkg/lifecycle/unfork/daemonless.go]: this can also be DI'd v.s. creating another instance

File: integration/lib.go (hunk 1 above)
Comment (marccampbell) [reply integration/lib.go]: Removed `modifyField`. Left this pulled out of the `CompareDir` for now, it was repeated, so /shrug. It's probably better to be out?

**Pull request merged**